lib/Backend/GlobOpt.cpp

```diff
@@ -7184,13 +7212,29 @@ GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
     if (prevSymStore && prevSymStore->IsStackSym() &&
         prevSymStore->AsStackSym()->HasByteCodeRegSlot())
     {
-        Assert(this->blockData.changedSyms != nullptr);
-        this->blockData.changedSyms->Set(prevSymStore->m_id);
+        this->SetChangedSym(prevSymStore->m_id);
     }
     valueInfo->SetSymStore(sym);
 }
 
 void
+GlobOpt::SetChangedSym(SymID symId)
+{
+    GlobOptBlockData * globOptData = &this->currentBlock->globOptData;
```

Comment:
>this->currentBlock [](start = 38, length = 18)

Let's just add a comment that currentBlock is the probably wrong block sometimes, but it just makes us overly conservative, not incorrect.